### PR TITLE
fix(budmetrics): improve ClickHouse column mapping for inference detail queries

### PR DESCRIPTION
**Pull request by dittops**
## Summary
- Fixed ClickHouse column mapping issues in inference detail queries by introducing helper methods for row-to-dict conversion
- Added `rows_to_dicts()` and `row_to_dict()` static methods to ClickHouseClient for named column access
- Refactored `get_inference_detail()` to use column names instead of positional indices, eliminating data shift errors
- Added comprehensive unit tests for the new helper methods

### ClickHouseClient Model (`models.py`)
- Added `rows_to_dicts()` static method for converting multiple query result rows to dictionaries
- Added `row_to_dict()` static method for converting single query result rows to dictionaries
- Both methods use cursor column descriptions to map column names to values
- Fixed linting issues with zip() strict parameter

### ObservabilityMetricsService (`services.py`)
- Refactored `get_inference_detail()` method to use named column access via `row_to_dict()`
- Replaced all positional index access (`row[0]`, `row[1]`, etc.) with named access (`row_dict.get('column_name')`)
- Improved error handling and logging with more descriptive column information
- Fixed gateway metadata parsing to handle column mapping correctly

### Tests (`test_clickhouse_client.py`)
- Added comprehensive test suite for new helper methods
- Tests cover basic conversion, empty inputs, single column scenarios, and complex data types
- Ensures proper handling of edge cases and data validation
- Added realistic GatewayAnalytics column structure test

## Test plan
- [x] Unit tests pass for new helper methods
- [x] Existing inference detail tests continue to work
- [x] Manual testing shows correct column mapping in inference queries
- [x] Gateway metadata parsing works correctly with named columns
- [x] All linting and formatting checks pass

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: services/budmetrics/budmetrics/observability/services.py [hunk 1]
```diff
@@ -974,205 +980,227 @@ async def get_inference_details(self, inference_id: str) -> EnhancedInferenceDet
...
+                        os_name=row_dict.get("os_name"),
+                        os_version=row_dict.get("os_version"),
+                        is_bot=row_dict.get("is_bot"),
+                        method=row_dict.get("method"),
+                        path=row_dict.get("path"),
+                        query_params=row_dict.get("query_params"),
+                        request_headers=safe_dict(row_dict.get("request_headers")),
+                        body_size=safe_int(row_dict.get("body_size")),
+                        api_key_id=row_dict.get("ga.api_key_id"),
+                        auth_method=row_dict.get("auth_method"),
```
[the remark this answers is not in the document]
Comment (dittops) [reply services/budmetrics/budmetrics/observability/services.py:1123]: @claude fix if this is relevant
